lib/redcord/server_scripts/shared/lua_helper_methods.erb.lua

```diff
@@ -30,3 +30,28 @@ local function set_list_intersect(set, list)
   end
   return set_intersect
 end
+
+-- Helper function to transform attribute values so that they become comparable as strings
+local function adjust_string_length(value)
+  if value == '' or value == nil then
+    return '!'
+  end
+  local whole_digits_count = 10
+  local decimal_digits_count = 4
```

Comment:
can we make this a lot longer to fit full `double` floats?  otherwise, we might want to verify the range is within our precision

https://en.wikipedia.org/wiki/Double-precision_floating-point_format


fwiw floats can get really big. maybe it's better to store their binary format here

Reply:
```ruby
def to_b(f)
  [f].pack('g').bytes.map{|n| "%08b" % n}.join
end

to_b(1.0) > to_b(0.5) # true
```
we need to handle the singed bit (the first bit) differently tho (0 -> `+`, 1 -> `-`)

Reply:
that means we'd need to store float differently, in their modified binary format (change the singed bit)

Reply:
alright, here's a Lua implementation:
```
function float2hexstr (n)
    if n == 0.0 then return "p00000000" end

    local sign = 0
    local prefix = 'p'
    if n < 0.0 then
        n = -n
        prefix = 'n'
    end

    local mant, expo = math.frexp(n)
    local hext = {}

    if mant ~= mant then
        hext[#hext+1] = string.char(0xFF, 0x88, 0x00, 0x00)

    elseif mant == math.huge or expo > 0x80 then
        if sign == 0 then
            hext[#hext+1] = string.char(0x7F, 0x80, 0x00, 0x00)
        else
            hext[#hext+1] = string.char(0xFF, 0x80, 0x00, 0x00)
        end

    elseif (mant == 0.0 and expo == 0) or expo < -0x7E then
        hext[#hext+1] = string.char(sign, 0x00, 0x00, 0x00)

    else
        expo = expo + 0x7E
        mant = (mant * 2.0 - 1.0) * math.ldexp(0.5, 24)
        hext[#hext+1] = string.char(sign + math.floor(expo / 0x2),
                                    (expo % 0x2) * 0x80 + math.floor(mant / 0x10000),
                                    math.floor(mant / 0x100) % 0x100,
                                    mant % 0x100)
    end

    return string.format(
      prefix .. "%x",
      tonumber(string.gsub(table.concat(hext),"(.)",
                                function (c) return string.format("%02X%s",string.byte(c),"") end), 16)
    )
end

a = float2hexstr(1.1)
b = float2hexstr(1.0)
print(a)
print(b)
print(a > b)

a = float2hexstr(123.45)
b = float2hexstr(0.12345)
print(a)
print(b)
print(a > b)

a = float2hexstr(0.123)
b = float2hexstr(-123.45)
print(a)
print(b)
print(a > b)

print(float2hexstr(0) > float2hexstr(-123))
print(float2hexstr(1) > float2hexstr(0))
```

Reply:
Let's implement this in the next iteration.